superset/db_engine_specs/base.py

```diff
@@ -1001,3 +1001,13 @@ def get_extra_params(database: "Database") -> Dict[str, Any]:
                 logger.error(ex)
                 raise ex
         return extra
+
+    @classmethod
+    def stop_queries(cls, username: str, dashboard_id: int) -> None:
```

Comment:
Can we name this `stop_dashboard_queries` since it accepts `dashboard_id` as a required parameter? Also, would `user_id` be better suited for this API than `username`?

Reply:
airbnb's internal API accept username and dashboard_id :)

Reply:
curious if there is an interest to have in in superset as well e.g. use smth like:
```CALL system.runtime.kill_query('20151207_215727_00146_tx3nr');```
https://github.com/prestodb/presto/issues/1515

However a challenge here would be tracking all queries and there ids

Reply:
correct. Dashboard is in synchronized mode, there is no query id passed from query engine to dashboard.

Reply:
if you find another solution, i am happy to learn :)

Reply:
Would it be possible to introduce a hook here that can either link a database or a db engine spec to a function that handles the query termination? Something similar to what `JINJA_CONTEXT_ADDONS` or `CUSTOM_TEMPLATE_PROCESSORS` does in `config.py`.